internal/pkg/dashboard/apidata/source.go

```diff
@@ -85,23 +82,36 @@ func (source *API) run(dataCh chan<- *data.Data) {
 }
 
 //nolint:gocyclo,cyclop
-func (source *API) gather() *data.Data {
-	result := &data.Data{
+func (source *Source) gather() *Data {
```

Comment:
Reverted API data gathering logic here to its previous shape, as we now don't need to do anymore an initial request to find out the set of nodes.